docs/library/openamp.rst

```diff
@@ -0,0 +1,104 @@
+:mod:`openamp` -- provides standard Asymmetric Multiprocessing (AMP) support
+============================================================================
+
+.. module:: openamp
+   :synopsis: provides standard Asymmetric Multiprocessing (AMP) support
+
+The openamp module provides a standard inter-processor communications infrastructure
+for MicroPython. The module handles all of the details of OpenAMP, such as setting up
+the shared resource table, initializing vrings etc.. and provides an API for using the
+RPMsg bus infrastructure with the `Endpoint` class. Furthermore, the module provides
+processor Life Cycle Management (LCM) support, such as loading firmware, starting and
+stopping a remote core, via the `RemoteProc` class.
+
+Example usage::
+
+    import openamp
+    
+    def ept_recv_callback(src_addr, data):
+        print("Received message on endpoint", data)
+    
+    # Create a new RPMsg endpoint to communicate with the remote core.
+    ept = openamp.Endpoint("vuart-channel", callback=ept_recv_callback)
+    
+    # Create a remoteproc object, load its firmware and start it.
+    rproc = openamp.RemoteProc("virtual_uart.elf") # Or entry point address (ex 0x081E0000)
+    rproc.start()
+    
+    while True:
+        if ept.is_ready():
+            ept.send("data")
+
+Functions
+---------
+
+.. function:: ns_callback(ns_callback)
+
+    Set the new service callback.
+
+        - ``ns_callback`` This function will be called back when the remote processor
+          announce new services. At the point the host processor can choose to create the
+          announced endpoint, if this particular service is supported, or ignore it if it's
+          not. If this function is not set, the host processor should first register the
+          endpoint locally, and it will be automatically bound when the remote announces
+          the service.
+
+Classes
+-------
+
+.. class:: Endpoint(name, callback, src_addr=ENDPOINT_ADDR_ANY, dst_addr=ENDPOINT_ADDR_ANY)
```

Comment:
I suggest calling the arguments just `src` and `dest`.  The fact they are part of the endpoint construction indicates they are addresses.